From cad60201fc2621a0e327d83a9bf5259c196e4b14 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?David=20Santamar=C3=ADa=20Rogado?= <howl.nsp@gmail.com>
Date: Tue, 17 Feb 2026 23:57:49 +0100
Subject: [PATCH] udev: rules: integration fix

ID_INTEGRATION is not being updated with hwdb entries, asign the new
value to it when hwdb has been imported.

We still need the 65-integration.rule assignment for devices that aren't
in hwdb.

While at it remove unneeded check in 70-touchpad.rules, as it was not
added for 70-joystick.rules with the statement if ID_INPUT_* is set and
ID_INPUT not, there is a bug elsewhere. And remove unneeded gotos in
both files.

Follow-up for a4381cae8bfacb1160967ac499c2919da7ff8c2b.

diff --git a/rules.d/70-joystick.rules b/rules.d/70-joystick.rules
index b80d203670..00e7ab49f3 100644
--- a/rules.d/70-joystick.rules
+++ b/rules.d/70-joystick.rules
@@ -6,7 +6,11 @@ KERNEL!="event*", GOTO="joystick_end"
 
 # joystick:<bustype>:v<vid>p<pid>:name:<name>:*
 KERNELS=="input*", ENV{ID_BUS}!="", \
-        IMPORT{builtin}="hwdb 'joystick:$env{ID_BUS}:v$attr{id/vendor}p$attr{id/product}:name:$attr{name}:'", \
-        GOTO="joystick_end"
+        IMPORT{builtin}="hwdb 'joystick:$env{ID_BUS}:v$attr{id/vendor}p$attr{id/product}:name:$attr{name}:'"
+
+# Spread the hwdb override to ID_INTEGRATION, in the future we could remove the
+# joystick hwdb entirely or retain it using the generic ID_INTEGRATION instead
+# specific ID_INPUT_JOYSTICK_INTEGRATION.
+ENV{ID_INPUT_JOYSTICK_INTEGRATION}!="", ENV{ID_INTEGRATION}="$env{ID_INPUT_JOYSTICK_INTEGRATION}"
 
 LABEL="joystick_end"
diff --git a/rules.d/70-touchpad.rules b/rules.d/70-touchpad.rules
index 7bede02dec..e51d8cf51a 100644
--- a/rules.d/70-touchpad.rules
+++ b/rules.d/70-touchpad.rules
@@ -1,13 +1,16 @@
 # do not edit this file, it will be overwritten on update
 
 ACTION=="remove", GOTO="touchpad_end"
-ENV{ID_INPUT}=="", GOTO="touchpad_end"
 ENV{ID_INPUT_TOUCHPAD}=="", GOTO="touchpad_end"
 KERNEL!="event*", GOTO="touchpad_end"
 
 # touchpad:<subsystem>:v<vid>p<pid>:name:<name>:*
 KERNELS=="input*", ENV{ID_BUS}!="", \
-        IMPORT{builtin}="hwdb 'touchpad:$env{ID_BUS}:v$attr{id/vendor}p$attr{id/product}:name:$attr{name}:'", \
-        GOTO="touchpad_end"
+        IMPORT{builtin}="hwdb 'touchpad:$env{ID_BUS}:v$attr{id/vendor}p$attr{id/product}:name:$attr{name}:'"
+
+# Spread the hwdb override to ID_INTEGRATION, in the future we could remove the
+# touchpad hwdb entirely or retain it using the generic ID_INTEGRATION instead
+# specific ID_INPUT_TOUCHPAD_INTEGRATION.
+ENV{ID_INPUT_TOUCHPAD_INTEGRATION}!="", ENV{ID_INTEGRATION}="$env{ID_INPUT_TOUCHPAD_INTEGRATION}"
 
 LABEL="touchpad_end"

